### ISSUE TITLE
Rewrite command module

### ISSUE THREAD

**Issue by matthias-ronge** [kitodo/kitodo-production#4495]
The command module should be rewritten. It has several code flaws:

- Only one character string can be transferred as a parameter via the interface. This is split hard into spaces in the module (also ignoring quotation marks) and transferred to the operating system as an array. Instead, the path to the command and an array with parameters should be transferred via the interface.
- An integer ID is transferred via the interface. This is generated randomly and is not used anywhere (dead code). This can be removed.
- The command module does not use multithreading for reading out return streams. That wouldn't work for commands that return a lot of text. If the system overflows the buffer, the command will cause a deadlock.

I find it questionable whether we need a module here. It seems more productive to me to use Apache Commons Exec (and understand it as the module) and thus simply execute the commands. But that, we should discuss this.

**Comment by henning-gerhardt** [comment]
Changing code to resolve this mentioned issues is good. But I'm unsure about your last point with the multithreading for reading out return statements. I don't know if this should be supported. If a called program produce a lot of output then the program shows some kind of error or something other is going wrong. Who should read and understand the output in this case? Maybe no one is "seeing" this output and maybe get only logged. But then it would be better that the executing of the program is failing because of to many output instead of displaying / logging this output.

Note to apache commons exec: last release is nearly 6 year ago (November 2014) with version 1.3 (reclaiming to compatible with Java 1.5!!!)  and no new release since them even they did commits in their source code repository. A replacement with a such old release should be good decided (I would disagree).

**Comment by matthias-ronge** [comment]
> I'm unsure about your last point with the multithreading for reading out return statements. I don't know if this should be supported. If a called program produce a lot of output then the program shows some kind of error or something other is going wrong.

Yes, this should be supported. Otherwise, the Java thread waiting for the script to finish will hang forever, which is a memory leak and may leave incomplete actions, like, from a script step, a subsequent database save of some objects in the code will never happen.

Especially if, as you point out, this might be an error case in the called script, there is no way to see the output, which you might want to have in that case to understand what happened. Apache Exec brings an `LogOutputStream` which can be set up to pass those messages to `kitodo.log` instead, and will guarantee to unblock the Java thread.

It also comes with a watchdog which can be set up to auto-kill the task if it has taken up too much time. I am a big fan of using timeouts everywhere, or, more poetically: _Even when the sun has turned into a red dwarf, it is still a long way to eternity._ How long do you really want to wait for a script call to return? I think, this should be configurable, and be enforced by the caller.

> Note to apache commons exec: last release is nearly 6 year ago (November 2014) with version 1.3 (reclaiming to compatible with Java 1.5

If the works is completed, the artist may retire. There is nothing bad about backward compatibility, by the way.

I can just say that I am using it since a decade now and it is very reliable.

The only drawback I ever found is its “guessing” of success on the exit code of the application (which is configurable, but anyhow) and throwing an Exception then. I would prefer it just to return the `int`. But that’s the only thing I ever came over; and by the way a script exiting with something else than `0` on success can be considered badly designed.

**Comment by BartChris** [comment]
There is now a more recent version of Commons Exec from 2024:
https://mvnrepository.com/artifact/org.apache.commons/commons-exec
